pkg/kubeapiserver/server/insecure_handler.go

```diff
@@ -39,7 +39,15 @@ func BuildInsecureHandlerChain(apiHandler http.Handler, c *server.Config) http.H
 	handler = genericapifilters.WithAudit(handler, c.AuditBackend, c.AuditPolicyChecker, c.LongRunningFunc)
 	handler = genericapifilters.WithAuthentication(handler, server.InsecureSuperuser{}, nil, nil)
 	handler = genericfilters.WithCORS(handler, c.CorsAllowedOriginList, nil, nil, nil, "true")
-	handler = genericfilters.WithTimeoutForNonLongRunningRequests(handler, c.LongRunningFunc, c.RequestTimeout)
+
+	// WithTimeoutForNonLongRunningRequests will call the rest of the request handling in a go-routine with the
+	// context with deadline. The go-routine can keep running, while the timeout logic will return a timeout to the client.
+	handler = genericfilters.WithTimeoutForNonLongRunningRequests(handler, c.LongRunningFunc)
+
+	// WithRequestDeadline sets a deadline for the request context appropriately
+	auditWrapper := genericapifilters.FailWithAuditWrapper(c.AuditBackend, c.AuditPolicyChecker)
+	handler = genericapifilters.WithRequestDeadline(handler, auditWrapper, c.LongRunningFunc, c.Serializer, c.RequestTimeout)
+
```

Comment:
why these empty lines? How are these two filters different?

Reply:
removed the empty lines